x/distribution/types/delegator_info.go

```diff
@@ -24,7 +26,13 @@ func NewDelegationDistInfo(delegatorAddr sdk.AccAddress, valOperatorAddr sdk.Val
 // Get the calculated accum of this delegator at the provided height
 func (di DelegationDistInfo) GetDelAccum(height int64, delegatorShares sdk.Dec) sdk.Dec {
 	blocks := height - di.DelPoolWithdrawalHeight
-	return delegatorShares.MulInt(sdk.NewInt(blocks))
+	accum := delegatorShares.MulInt(sdk.NewInt(blocks))
+
+	// defensive check
+	if accum.IsNegative() {
+		panic(fmt.Sprintf("negative accum: %v\n", accum.String()))
```

Comment:
I think it might be more helpful to panic with a string containing the height, delegator pool withdrawal height, and delegator shares.

Reply:
added